bxsolana/provider/ws.py

```
@@ -109,3 +113,12 @@ def ws_devnet() -> Provider:
 
 def ws_local() -> Provider:
     return WsProvider(endpoint=constants.LOCAL_API_WS)
+
+def _validate_response(response: Any, response_type: Type["T"]):
+    if not isinstance(response, dict):
+        raise Exception(f"response {response} was not a dictionary")
+
+    message = response_type().from_dict()
+    d = message.to_dict()
+    if len(d) == 0:
+        raise Exception(f"response {response} was not of type {T}")
```

Actually you should probably capture the error specific type message, like `HttpError`. In that case you want to raise the error message as the exception, instead of a generic `response was not of type`. Also, `T`, should be `response_type` in this error message.

---

Do you mean a `message` field in the error? I haven't seen this error before, so I am basing it on what you wrote in the above comment